Gravatar/MruCache.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Collections.Generic;
+using JetBrains.Annotations;
+
+namespace Gravatar
+{
+    // TODO consider moving this type (and associated test fixture) to a more general assembly
+
+    /// <summary>
+    /// Associative cache with fixed capacity that expires the last used item first.
+    /// </summary>
+    /// <typeparam name="TKey">Type of keys in the cache.</typeparam>
+    /// <typeparam name="TValue">Type of values in the cache</typeparam>
+    public sealed class MruCache<TKey, TValue>
```

Comment:
:memo: This type is not thread safe, so you'll need to synchronize either in the methods or in the callers.

Reply:
My sense is that it's only called on the UI thread. Will double check. Can add assertions to that effect.

Reply:
Can't guarantee it's UI only. Synchronised the caller.